source/js/main.js

```diff
@@ -496,34 +574,97 @@ const createRecipeClicked = () => {
   console.log(finalObject);
 };
 
-/**
- * active when show more button click
- * add # more recipe cards by fetch # more recipe from API with offset
- * populate more recipe cards
- */
 async function showMoreClicked() {
   const query = getSearchQuery();
   const searchResultsContainer = document.getElementById('search-results-container');
-
   const numOfCardExist = searchResultsContainer.childElementCount;
   const numOfAdditionRecipeCards = DEFAULT_NUM_CARDS;
 
+  const sorts = getSortKey();
+  const ordering = getOrderingKey();
+  const mealType = getMealTypeKey();
+  const dietS = getDietKey();
+  const maxPrepTime = getMaxPrepTime();
+  const cuisineString = getCuisinesKeys();
+  const intoleranceString = geIntolerances();
+
   const searchResult = await apiFuncs.getRecipesByName(
     query,
-    numOfAdditionRecipeCards,
+    DEFAULT_NUM_CARDS,
     numOfCardExist,
+    {
+      sort: sorts,
+      sortDirection: ordering,
+      cuisine: cuisineString,
+      type: mealType,
+      diet: dietS,
+      intolerances: intoleranceString,
+      maxReadyTime: maxPrepTime,
+    },
   );
-  storageFuncs.storeRecipeData(query, searchResult);
+  console.log(searchResult);
+
+  const storeName = query + sorts + ordering + cuisineString + mealType + dietS
+  + intoleranceString + maxPrepTime;
+  storageFuncs.storeRecipeData(storeName, searchResult);
 
   const numOfRecipe = numOfAdditionRecipeCards + numOfCardExist;
   const localCategories = JSON.parse(localStorage.getItem('explore-categories'));
 
-  for (let i = numOfCardExist; (i < numOfRecipe) && (i < localCategories[query].length); i += 1) {
-    const singleResultRecipeId = localCategories[query][i];
+  for (let i = numOfCardExist; (i < numOfRecipe) && (i < localCategories[storeName].length);
+    i += 1) {
+    const singleResultRecipeId = localCategories[storeName][i];
     createRecipeCards([singleResultRecipeId], searchResultsContainer, 1);
   }
 }
 
+async function applyClicked() {
+  const query = getSearchQuery();
+  const searchResultsContainer = document.getElementById('search-results-container');
+
+  // get filter and sort info from interface
+  const sorts = getSortKey();
+  const ordering = getOrderingKey();
+  const mealType = getMealTypeKey();
+  const dietS = getDietKey();
+  const maxPrepTime = getMaxPrepTime();
+  const cuisineString = getCuisinesKeys();
+  const intoleranceString = geIntolerances();
+
+  console.log(sorts);
```

Comment:
Console logs should  be removed ASAP. I'll approve, but remember to do this ASAP or make an issue for it.

Reply:
removed